### ISSUE TITLE
Almost completely awesome! but...

### ISSUE THREAD

**Issue by michael-vansickle** [sedwards2009/dojo-typescript-class#1]
Hello,

I have been playing around with this technique and agree that it is, in most ways, superior to the one that I had come up with. Only one problem that I see: for templated widgets, I can't find an order that instantiates the widget correctly.

The canonical way to try this would be something like: 

``` ts
///<amd-dependency path="dojo/text!./_templates/template.html" />
/// <reference path="../../typings/dijit.d.ts" />

import dojoClass = require("app/dojoClass");
import _WidgetBase = require("dijit/_WidgetBase");
import _TemplatedMixin = require("dijit/_TemplatedMixin");

var template: string = require("dojo/text!./_templates/RulesView.html");

class Foo extends dijit._WidgetBase {
    templateString: string = template;

    constructor(params?: Object, elem?: HTMLElement) {
        super();
        dojoClass.construct(this, params, elem);
    }
}

dojoClass.create(Foo, [_WidgetBase, _TemplatedMixin]);

export=Foo;
```

however, this causes the application to try to new up the super (dijit._WidgetBase) and the new widget. This causes Dojo to throw an error since it tries to double instantiate this widget and assign them both the same ID. 

If we can find a way around this, then I think it addresses all of the use cases in a much more efficient way. I'll keep working on it and update this issue if I find anything.

**Comment by sedwards2009** [comment]
Yes, that situation where Dojo retries to re-new() the widget multiple times sounds familiar and I had to battle it for quite some time to get this far.

Mixins and multiple inheritance is the next battle. If you make progress first then let me know. :-)

**Comment by sedwards2009** [comment]
Just an update. I've been chipping away at these problems on and off for the last week.

Dojo doesn't chain constructors, but TS does. So we have to prevent super() in a TS subclass from executing the constructor from its Dojo based superclass. I've now got a 'wedge' in front of Dojo's declare() so that any call to declare() goes through my code. My code then wraps the constructor function in a new function where I can check for, and stop calls from subclasses to Dojo constructor functions. That constructor wrapping seems to mostly work now.

The next problem that I've been able to sniff out is that if you make a template based Dijit with a 'templateString' property, then Dojo expects this property to be on the prototype which instances of your custom Dijit will point to. But in TS if you have a field called 'templateString' then it will only appear inside your instance object and not in the prototype which belongs to the class. Working on this.

Those are the highlights modulo a bunch of other smaller problems which screwed things up along the way.
